roms_tools/utils.py

```diff
@@ -0,0 +1,125 @@
+from numbers import Integral
+
+import numpy as np
+import xarray as xr
+
+
+def partition(ds: xr.Dataset, nx: int = 1, ny: int = 1) -> list[xr.Dataset]:
+    """
+    Split a ROMS dataset up into nx by ny spatial tiles.
+    """
+
+    if not isinstance(nx, Integral) or not isinstance(ny, Integral):
+        raise ValueError("nx and ny must be integers")
+
+    # if eta is periodic there are no ghost cells along those dimensions
+    if ds.sizes["eta_rho"] == ds.sizes["eta_v"]:
+        n_eta_ghost_cells = 0
+    else:
+        n_eta_ghost_cells = 1
+
+    # if xi is periodic there are no ghost cells along those dimensions
+    if ds.sizes["xi_rho"] == ds.sizes["xi_u"]:
+        n_xi_ghost_cells = 0
+    else:
+        n_xi_ghost_cells = 1
+
+    def integer_division_or_raise(a: int, b: int) -> int:
+        remainder = a % b
+        if remainder == 0:
+            return a // b
+        else:
+            raise ValueError(
+                f"Partitioning nx = {nx} ny = {ny} does not divide the domain into subdomains of integer size."
+            )
+
+    eta_rho_domain_size = integer_division_or_raise(
+        ds.sizes["eta_rho"] - 2 * n_eta_ghost_cells, nx
+    )
+    eta_v_domain_size = integer_division_or_raise(
+        ds.sizes["eta_v"] - 1 * n_eta_ghost_cells, nx
+    )
```

Comment:
Not all ROMS input datasets will have the `eta_v` dimension. In fact, the only dimensions that will always be there are `eta_rho` and `xi_rho`. So we should make the partitioned sizes conditioned on whatever dimensions are present.

Reply:
Addressed in https://github.com/CWorthy-ocean/roms-tools/pull/101/commits/981cb46981d94e57d8bb1a0504b086b9e95ed48c